### ISSUE TITLE
[DataGrid] Grid resizes itself automatically in the infinite loop

### ISSUE THREAD

**Issue by vsDizzy** [mui/mui-x#11863]
### Search keywords

resize loop

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

This bug is reproducible in Chrome or Edge
[Live example](https://codesandbox.io/p/sandbox/naughty-blackburn-8lzwt2)

### Current behavior

Grid resizes to the right infinitely

### Expected behavior

Grid stays in place

**Comment by evgeniiarts** [comment]
@vsDizzy Pass some rows to your data grid, as without them it doesn't know how long the content of you flex column shall have and resizes infinitely.
https://codesandbox.io/p/sandbox/great-fire-wx9cnx

**Comment by vsDizzy** [comment]
Yes, this is the case. We have a search filter and when no data is found this bug happens.
And this is the case where grid should display "No data" right?
